plugins/ckan_operators/resource_operator.py

```diff
@@ -9,53 +9,186 @@
 
 class GetOrCreateResourceOperator(BaseOperator):
     """
-    Returns CKAN resource object, or creates it if it does not exist 
+    Returns CKAN resource object, or creates it if it does not exist
+
+    Expects as inputs:
```

Comment:
so there are 2 ways to put in most inputs. Each unique input is described there, and a description that there are 2 ways (ie: 2 sets of actual instance inputs) is described in that last line of the docstring.

TBH though I dont like it, and since we'll have to describe this "to-ways-to-enter-your-inputs" thing for lots of operators, I still need to think of a smarter fix.

Reply:
This is certainly one of the limitations around this approach, could be confusing. Something we'll need to balance for sure.

I think it's best to list every single input the function takes, mark them as required/optional, and capture in the description of the input if it's either this input or another one but not both.

That way at least we communicate it is an intentional design decision rather than a mistake, which I think would lead to more confusion.

Reply:
Future PR